nix_update/eval.py

```diff
@@ -142,14 +142,23 @@ def eval_expression(
     return f"""
 let
   {indent(dedent(let_bindings), "  ")}
+  positionFromMeta = pkg: let
+    parts = builtins.match "(.*):([0-9]+)" pkg.meta.position;
+  in if pkg ? meta.position then {{
+    file = builtins.elemAt parts 0;
+    line = builtins.fromJSON (builtins.elemAt parts 1);
+  }} else null;
+
   raw_version_position = sanitizePosition (builtins.unsafeGetAttrPos "version" pkg);
 
   position = if pkg ? isRubyGem then
     raw_version_position
   else if pkg ? isPhpExtension then
     raw_version_position
-   else
-    sanitizePosition (builtins.unsafeGetAttrPos "src" pkg);
+  else if (builtins.unsafeGetAttrPos "src" pkg) != null then
+    sanitizePosition (builtins.unsafeGetAttrPos "src" pkg)
+  else
+    sanitizePosition (positionFromMeta pkg);
```

Comment:
If we have meta.position, should it not be preferred over `src`?

Reply:
i was trying to change as little of existing behaviour as possible, by if you're on board with this change then why not.